### PR TITLE
Categorize HDFS read error for ORC

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by martint** [comment]
looks good

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java [hunk 1]
```diff
@@ -41,6 +45,15 @@ public void close()
...
     {
-        inputStream.readFully(position, buffer, bufferOffset, bufferLength);
+        try {
+            inputStream.readFully(position, buffer, bufferOffset, bufferLength);
+        }
+        catch (PrestoException e) {
+            // just in case there is a Presto wrapper or hook
+            throw e;
+        }
+        catch (Exception e) {
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java]: This should be `IOException | RuntimeException`

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java [hunk 2]
```diff
@@ -41,6 +45,15 @@ public void close()
...
+            throw new PrestoException(HiveErrorCode.HIVE_FILESYSTEM_ERROR, format("HDFS error reading from %s at %s", this, position), e);
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java]: Static import error code

Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java]: Say "at position %s"

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java (hunk 1 above)
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/orc/HdfsOrcDataSource.java]: Actually, not sure we should catch an arbitrary `RuntimeException` here

**Pull request closed without merging**